pkg/schema/converter/k8stonorman.go

```diff
@@ -38,13 +40,11 @@ func GVRToPluralName(gvr schema.GroupVersionResource) string {
 	return fmt.Sprintf("%s.%s", gvr.Group, gvr.Resource)
 }
 
+// ToSchemas creates the schemas for a K8s server, using client to discover groups/resources, and crd to potentially
+// add additional information about new fields/resources. Mostly ties together AddDiscovery and AddCustomResources
 func ToSchemas(crd v1.CustomResourceDefinitionClient, client discovery.DiscoveryInterface) (map[string]*types.APISchema, error) {
 	result := map[string]*types.APISchema{}
 
-	if err := AddOpenAPI(client, result); err != nil {
```

Comment:
Can you clarify why this was removed - I guess so that resourceFields would be empty? It looks like now the description is also missing because of this, that seems like a slight drawback.

Reply:
This part mostly gave recursive definition for the core types from what I can see on testing (i.e. what types are in a pod spec).  Since we no longer need to define those values, I think it's safe to remove, though I can look at it again if there's anything that you think it had that we need (I can also send over a with/without diff that may help if you want to see what fields are missing in it).

As far as the description goes, it looks like (from hitting `/openapi/v2` directly) our core types wouldn't have a description anyway - near as I can tell both the well-defined provisioning.cattle.io cluster and the less well-defined management.cattle.io cluster objects are missing a description field. I guess this does leave out these fields for the core types, but since you can use k8s for this information I don't think this is a large loss. I agree with you that in an optimal world this would be there for us to consume later, but I'd guess that we can pick that up from the CRD definition (which we call separately), so I'd prefer to look there. 

I can re-add and adjust this, but I would ask that we consider if that type of a change (and writing tests for all of that) is worth it just for the description field.

Reply:
Took another look at this. While most of this does appear to be the recursive core type definition, there is at least one resource that concerns me - the `apigroups` resource. This resource serves to list all of the apigroups available on the server (like `core` or `apps`) and their associated versions. It looks like most of the logic which generates this resource is in another place as a template - see [here](https://github.com/rancher/steve/blob/a10fe811f58ff7d92ffee51dd3f11010afbaf115/pkg/resources/apigroups/apigroup.go#L14). However, this template won't get applied, and the resource won't be created unless a schema exists, see [here](https://github.com/rancher/steve/blob/a10fe811f58ff7d92ffee51dd3f11010afbaf115/pkg/schema/collection.go#L115-L126). I'm unsure if the UI uses this value, but to me it would seem safer to include it rather than exclude it and hope everything goes ok.

In short, I missed something with one resource, and I'll see if I can fix it by re-introducing the top level resources here as suggested.

Reply:
I looked at this with Micahel, and `apigroups` is not a Kubernetes Resource Type. 
It is just distilled information that can be obtained by parsing the output of `/v1/schemas`

We should not include this extra functionality if the following statements stand true.
1. We desire to keep Steve as lightweight as possible. (AKA only add necessary functionality).
2. The endpoint is not currently needed by the UI.